### ISSUE TITLE
test(testing): test_train_eval needs cfg_eval.model + test/acc assertion aligned with ksin/ffn

### ISSUE THREAD

**Issue by ktinubu** [tinaudio/synth-setter#624]
## Context

Follow-up from [#623](https://github.com/tinaudio/synth-setter/pull/623) (which fixes #617 for the `FileNotFoundError` on `surge_mini`). While reviewing that PR, Copilot pointed out that the test's train->eval roundtrip is still incomplete even after aligning `cfg_eval.data` with `cfg_train.data`.

## Problems

After `cfg_eval.data = cfg_train.data` (ksin), `cfg_eval` still composes from `configs/eval.yaml` defaults:

- `model: surge_flow` -> `src.models.surge_flow_matching_module.SurgeFlowMatchingModule`
- `callbacks: eval_surge`

But `train(cfg_train)` produces an `ffn`/`KSinFeedForwardModule` checkpoint. In `src/eval.py::evaluate`, Lightning calls `trainer.test(model=model, ..., ckpt_path=cfg.ckpt_path)` which will attempt to load the ffn state_dict into the instantiated `surge_flow` model — this will fail on mismatched state_dict keys.

Additionally, `tests/test_eval.py:48` asserts:

```python
assert test_metric_dict["test/acc"] > 0.0
```

But neither `ksin_ff_module.test_step` nor `surge_flow_matching_module.test_step` logs `test/acc`. Only `mnist_module` does. The assertion is a stale artifact from when the default model was MNIST.

## Proposed fix

In `tests/test_eval.py::test_train_eval`:

1. Override `cfg_eval.model = cfg_train.model` and `cfg_eval.callbacks = cfg_train.callbacks` (or pick a matching eval-specific callbacks config) so the instantiated eval model matches the trained checkpoint.
2. Replace `test/acc` assertion with a metric both modules produce, e.g. `test/param_mse` or `test/loss` (assert it's a finite, non-negative float).

## Scope

- Scoped strictly to `tests/test_eval.py::test_train_eval`.
- Do not modify production configs (`configs/eval.yaml`, `configs/data/surge_mini.yaml`, model/callback configs).

## Parent

Part of #155 (Phase 2: Coverage & stability) -> #149 (Epic: test infrastructure & coverage).

Refs #623
Refs #617
